lib/psm.c

```diff
@@ -511,13 +511,14 @@ void rpmpsmNotify(rpmpsm psm, int what, rpm_loff_t amount)
  */
 static void markReplacedInstance(rpmts ts, rpmte te)
 {
+    /* this must match rpmNameVersionCompare in depends.c */
```

Comment:
No, rpm currently does not use the sha sig to test if this is a "reinstall" or an update/downgrade. The weird update/reinstall difference is a mess but it seems that it has to stay for compatibility reasons. This is about the addSelfErasures code.

When you have foo-1-1.i586 installed and you do 'rpm -U foo-1-1.i686', rpm will not add an erase element for the i586 package.